### ISSUE TITLE
4x6 label was parsed to 9 pages before printing

### ISSUE THREAD

**Issue by cindyx99** [qzind/tray#905]
A label sized 4" x 6" was parsed to 9 pages of PDF before sending the PDFwriter. I tried to print to a desktop printer and the print queue shows 9 pages as well. Would someone please help look into this? 

Version of qz tray: 2.2.1 for Mac. 
Here is the log file:

[DEBUG] 2022-01-09T15:00:22,149 @ qz.printer.PrintServiceMatcher:-1
	Found match: PDFwriter
[WARN] 2022-01-09T15:00:22,150 @ qz.printer.PrintOptions:-1
	Raw option "altPrinting" is deprecated.  Please use "forceRaw" instead.
[DEBUG] 2022-01-09T15:00:22,247 @ qz.printer.action.PrintImage:-1
	Parsed 1 images for printing
[TRACE] 2022-01-09T15:00:22,251 @ qz.printer.action.PrintPixel:-1
	DPI: [300.0x0.0]	CNV: 72.0
[TRACE] 2022-01-09T15:00:22,252 @ qz.printer.action.PrintPixel:-1
	Drawable area: 1.0,1.0:7.5,10.0
[TRACE] 2022-01-09T15:00:22,252 @ qz.printer.action.PrintPixel:-1
	[one-sided, 30000x30000 dphi, (25.4,25.4)->(190.5,254.0)mm, color]
[TRACE] 2022-01-09T15:00:22,252 @ qz.printer.action.PrintImage:-1
	Image to be printed across 9 pages
[INFO] 2022-01-09T15:00:22,254 @ qz.printer.action.PrintPixel:-1
	Starting printing (1 copies)
[TRACE] 2022-01-09T15:00:22,276 @ qz.printer.action.PrintImage:-1
	Requested page 0 for printing
[TRACE] 2022-01-09T15:00:22,277 @ qz.printer.action.PrintImage:-1
	Requested page 0 for printing
[DEBUG] 2022-01-09T15:00:22,278 @ qz.printer.action.PrintImage:-1
	Paper area: 72,72:522,680
[TRACE] 2022-01-09T15:00:22,278 @ qz.printer.action.PrintImage:-1
	Image size: 129.6,172.79999999999998

**Comment by tresf** [comment]
Please attach the file here for review.  If it contains sensitive data and you are a Premium Support subscriber, you may email it to support@qz.io instead.

**Comment by cindyx99** [comment]
Label to be printed
![test_label](https://user-images.githubusercontent.com/36333994/148706033-a91e7d4e-8ff9-4aca-99a9-35c34907f705.png)

PDF generated:
[job_30 QZ Tray Pixel Print.pdf](https://github.com/qzind/tray/files/7836212/job_30.QZ.Tray.Pixel.Print.pdf)

**Comment by tresf** [comment]
> Label to be printed

@cindyx99 I see two attachments, one is a PNG preview, the other is the output from QZ Tray.

Can you please send us the PDF in unmodified form, before QZ Tray prints it?

Also, a copy of the logs will help us understand why it's being chopped up.  To get the logs, Advanced, Diagnostics, Zip Logs to Desktop.

**Comment by cindyx99** [comment]
Hi,

Thanks for following up. 
The input was the png file. PDF was a capture of what QZ Tray was trying to print.
Attached is the log file for your reference. Thank you.
[qz-2022-01-10_1712.zip](https://github.com/qzind/tray/files/7842210/qz-2022-01-10_1712.zip)

**Comment by lite1979** [comment]
@cindyx99 

The configuration needs to match the printer's hardware. I don't know if you can configure this through Chit Chat's website or not, but if you cannot, you will need to escalate with them.

The first thing I did was download the .png file and look at it's properties. It's 1200 pixels wide, and 1800 pixels tall, so it is proportionate to a 4x6 inch label.

However, the config for your HP printer makes it print across several pages; I will explain below


`2022-01-08T22:50:25,353 [DEBUG] Message: {"call":"print","promise":{},"params":{"printer":{"name":"HP Officejet 5740 series"},"options":{"colorType":"color","copies":1,` < safe to ignore

`"density":"300",` < this is the first problem. Specifying density on a printer can cause scaling issues when the software is trying to get the print job right pixel-for-pixel _and_ autoScaling is turned off

based on this [spec page](https://support.hp.com/us-en/product/hp-officejet-5740-e-all-in-one-printer-series/6617253/document/c05031679), the print density is "up to 1200dpi"  - your printer driver will tell you more, but we rarely see an inkjet or laserjet printer with a dpi of 300. _It's probably best to leave this value blank_


`"duplex":false,"fallbackDensity":600,"interpolation":"bicubic","jobName":null,` < safe to ignore

`"margins":{"top":"1","left":"1"},` < this may not be necessary, but if you're printing a 4x6 image to 8.5x11 paper, it's probably good to have at least a quarter inch margin on the top and the left so nothing gets cut off in the unprintable area near those edges.

`"orientation":null,"paperThickness":null,"printerTray":null,` < safe to ignore

`"rasterize":true,`< rasterization should be false here; it will also use less memory and print faster :)

`"rotation":0,` < safe to ignore

`"scaleContent":false,` < this should be set to true, _but_ you will want the size parameter to be 4x6, not 8.5x11

`"size":{"width":8.5,"height":11},` < as mentioned above, if you want the printed result to occupy a 4x6 area on the 8.5x11 paper, you will want to change this to 4x6 and set scaleContent to "true"

To summarize, you will want to make the following changes to your configuration:

density: "" (null)
margins: top: .25, left: .25 (or whatever you're comfortable with; I imagine this is going into a clear plastic shipping envelope, so as long as it's easy to see and scan)
rasterize:  false
scaleContent: true
size: width: 4, height, 6

I'm going to close this issue as it does not appear to be a bug with our software, but I would encourage you to escalate any issues with Chit Chats before re-opening.

**Comment by cindyx99** [comment]
Thank you very much. 
One last question: where do I change the configuration?

**Comment by lite1979** [comment]
@cindyx99 As I mentioned in my comment, you will have to escalate this with Chit Chats. We did not author their website.

**Comment by cindyx99** [comment]
Thank you. That answered my question. I was not sure if the configuration can be changed from my own printer setting.
